addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -0,0 +1,323 @@
+                'we-selection-items[name="link_style_shape"] > we-button',
+            ];
+            for (const option of this.$(options.join(','))) {
+                const $option = $(option);
+                const value = $option.data('value');
+                let active = false;
+                if (value) {
+                    const subValues = value.split(',');
+                    let subActive = true;
+                    for (let subValue of subValues) {
+                        const classPrefix = new RegExp('(^|btn-| |btn-outline-)' + subValue);
+                        subActive = subActive && classPrefix.test(this.data.iniClassName);
+                    }
+                    active = subActive;
+                } else {
+                    active = !this.data.iniClassName.includes('btn-');
+                }
+                this._setSelectOption($option, active);
+            }
+        }
+        if (this.data.url) {
+            var match = /mailto:(.+)/.exec(this.data.url);
+            this.$('input[name="url"]').val(match ? match[1] : this.data.url);
+            this._onURLInput();
+        }
+        if (this.data.isNewWindow) {
+            this.$('we-button.o_we_checkbox_wrapper').toggleClass('active', true);
+        }
+
+        this._updateOptionsUI();
+        this._adaptLink(false);
+
+        this.$('input:visible:first').focus();
+
+        return this._super.apply(this, arguments);
+    },
+    destroy: function () {
+        this.options.wysiwyg.odooEditor.automaticStepSkipStack();
+        $('.oe_edited_link').removeClass('oe_edited_link');
+        const $contents = this.$link.contents();
+        if (!this.$link.attr('href') && !this.colorCombinationClass) {
+            $contents.unwrap();
+        }
+        this.$button.removeClass('active');
+        const start = $contents[0] || this.$link[0];
+        const end = $contents[$contents.length - 1] || this.$link[0];
+        setCursor(start, 0, end, nodeSize(end));
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Adapt the link to changes.
+     *
+     * @private
+     */
+    _adaptLink: function (createStep = true) {
+        var data = this._getData();
+        if (data === null) {
+            return;
+        }
+        const attrs = Object.assign({}, this.data.oldAttributes, {
+            target: data.isNewWindow ? '_blank' : '',
+            href: data.url,
+            class: `${data.classes}`,
+        });
+        const $links = $('.oe_edited_link');
+        $links.removeClass('oe_edited_link');
+        this.$link.attr(attrs);
+        if (data.label !== this.data.originalText || data.url !== this.data.url) {
+            const label = (data.label && data.label.length) ? data.label : data.url;
+            this.$link.html(label);
```

Comment:
This still doesn't look right, `label` is clearly not always html. Easy to test: type `<b>this text is not bold</b>` in the editor, select it and then hit Ctrl-K, the text gets rendered in bold, showing that text is being confused with markup.
This should even be covered by a test, no?

Reply:
We are now sanitizing the label (now renamed `content`) but there is still possible images in the content so we need to use `.html()`.